tools/cldr-code/src/main/java/org/unicode/cldr/util/ExtraPaths.java

```diff
@@ -1,100 +1,82 @@
 package org.unicode.cldr.util;
 
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
 
 public class ExtraPaths {
 
-    private static final Map<NameType, ExtraPaths> instances = new HashMap<>();
-
-    public static synchronized ExtraPaths getInstance(NameType nameType) {
-        return instances.computeIfAbsent(nameType, ExtraPaths::new);
+    public static void add(Collection<String> toAddTo) {
+        toAddTo.addAll(SingletonHelper.INSTANCE.paths);
     }
 
-    private final NameType nameType;
-    private final Collection<String> paths;
-
-    private ExtraPaths(NameType nameType) {
-        this.nameType = nameType;
-        paths = new HashSet<>();
+    private static class SingletonHelper {
+        private static final Singleton INSTANCE = new Singleton();
     }
 
-    void append(Collection<String> toAddTo) {
-        if (paths.isEmpty()) {
-            populatePaths();
-        }
-        toAddTo.addAll(paths);
-    }
+    private static class Singleton {
+        private final Collection<String> paths;
 
-    private void populatePaths() {
-        // TODO: https://unicode-org.atlassian.net/browse/CLDR-17014
-        // StandardCodes.CodeType codeType = StandardCodes.CodeType.fromNameType(nameType);
-        // See https://github.com/unicode-org/cldr/pull/4287
-        StandardCodes.CodeType codeType;
-        switch (nameType) {
-            case LANGUAGE:
-                codeType = StandardCodes.CodeType.language;
-                break;
-            case SCRIPT:
-                codeType = StandardCodes.CodeType.script;
-                break;
-            default:
-                throw new IllegalArgumentException("TODO: CodeType.fromNameType");
+        Singleton() {
+            paths = new HashSet<>();
+            addPaths(NameType.SCRIPT);
+            addPaths(NameType.LANGUAGE);
```

Comment:
At the end of this method, you need to make paths be an ImmutableSet. Add an extra parameter to addPaths, etc. and change this to:
```suggestion
            Set<String> temp = new TreeSet(); // might as well put them in order; easier to debug
            addPaths(NameType.SCRIPT, temp);
            addPaths(NameType.LANGUAGE, temp);
            paths = ImmutableSet.copyOf(temp); // preserves order (Sets.copyOf doesn't)
```